controllers/serviceaccount_controller.go

```diff
@@ -184,7 +182,8 @@ func (r *ServiceAccountReconciler) Reconcile(ctx context.Context, req reconcile.
 	}()
 
 	if !vsphereCluster.DeletionTimestamp.IsZero() {
-		return ctrl.Result{}, r.reconcileDelete(ctx, clusterContext)
+		controllerutil.RemoveFinalizer(clusterContext.VSphereCluster, vmwarev1.ProviderServiceAccountFinalizer)
```

Comment:
I assume it's okay that someone else is responsible to cleanup existing ConfigMaps? (I think it's fine, just want to double check)

Reply:
Looks like we never created it, so I think it should be okay that "whoever creates it" should clean it up once we don't use it anymore

Reply:
What we could do though is deleting it if the env vars are set and it still exists.

Reply:
I would probably prefer if we just tell people to delete it. The way some people upgrade (jumping over various versions) we have to keep this forever otherwise